### PR TITLE
12 image conversion

### LINKED ISSUE #12: Image conversion
As a developer I want service, that automatically resizes, compresses and converts  an image, so that it can be stored by the Person management service

Estimated: 5+h

### PR DESCRIPTION
**Pull request by AndHager**
Closes #12

### CODE REVIEW

File: brainkeeper/package.json [hunk 1]
```diff
@@ -20,9 +20,12 @@
     "@angular/material": "^8.2.3",
     "@angular/platform-browser": "~8.2.9",
     "@angular/platform-browser-dynamic": "~8.2.9",
-
     "@angular/router": "^8.2.14",
+    "blueimp-canvas-to-blob": "^3.17.0",
+    "compress.js": "^1.1.2",
```
Comment (walli545) [inline brainkeeper/package.json]: Do you use these libraries? If not please remove them.

File: brainkeeper/src/index.html [hunk 2]
```diff
@@ -12,5 +12,7 @@
 </head>
 <body>
   <app-root></app-root>
+  var remote = require('electron').remote
```
Comment (walli545) [inline brainkeeper/src/index.html]: I think this can be removed

File: brainkeeper/tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
```
Comment (walli545) [inline brainkeeper/tsconfig.json]: Is this required?

File: brainkeeper/src/app/services/image-service.service.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        resolve(this.changeImage(img, file.type === 'jpg' ? 'jpeg' : 'png', width, height, quality));
+      };
+      reader.onerror = () => {
+        throw new Error('In the image conversion went something wrong.');
+      };
+    });
+  }
+
+  changeImage(
+    img: HTMLImageElement, type: string, width: number, height: number, quality: number
```
Comment (walli545) [inline brainkeeper/src/app/services/image-service.service.ts]: Width and height arguments are not used inside the method, should they be removed?

File: brainkeeper/src/app/services/image-service.service.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        if (width === 0) {
+          width = img.width;
+        }
+        if (height === 0) {
+          height = img.height;
+        }
+        resolve(this.changeImage(img, file.type === 'jpg' ? 'jpeg' : 'png', width, height, quality));
+      };
+      reader.onerror = () => {
+        throw new Error('In the image conversion went something wrong.');
```
Comment (walli545) [inline brainkeeper/src/app/services/image-service.service.ts]: the promise should be rejected on error cases, please use reject(error)

File: brainkeeper/src/app/services/image-service.service.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class ImageServiceService {
```
Comment (walli545) [inline brainkeeper/src/app/services/image-service.service.ts]: Please rename to ImageService, also rename file to image.service.ts

File: brainkeeper/tsconfig.json (hunk 3 above)
Comment (AndHager) [reply brainkeeper/tsconfig.json]: no it is not.

File: brainkeeper/src/index.html (hunk 2 above)
Comment (AndHager) [reply brainkeeper/src/index.html]: Yes it can be removed.

File: brainkeeper/package.json (hunk 1 above)
Comment (AndHager) [reply brainkeeper/package.json]: No i don't use them.

File: brainkeeper/src/app/services/image-service.service.ts (hunk 5 above)
Comment (AndHager) [reply brainkeeper/src/app/services/image-service.service.ts]: Ok, thank you.

File: brainkeeper/src/app/services/image-service.service.ts (hunk 6 above)
Comment (AndHager) [reply brainkeeper/src/app/services/image-service.service.ts]: Done.

File: brainkeeper/src/app/services/image-service.service.ts (hunk 4 above)
Comment (AndHager) [reply brainkeeper/src/app/services/image-service.service.ts]: No they are now used to crop the image.

**Pull request merged**